subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/CrossProjectModelAccessTrackingParentDynamicObject.kt

```diff
@@ -0,0 +1,139 @@
+        return delegate.hasMethod(name, *arguments)
+    }
+
+    override fun tryInvokeMethod(name: String?, vararg arguments: Any?): DynamicInvokeResult {
+        onAccess(MemberKind.METHOD, name)
+        return delegate.tryInvokeMethod(name, *arguments)
+    }
+
+    override fun hasProperty(name: String?): Boolean {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.hasProperty(name)
+    }
+
+    override fun tryGetProperty(name: String?): DynamicInvokeResult {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.tryGetProperty(name)
+    }
+
+    override fun trySetProperty(name: String?, value: Any?): DynamicInvokeResult {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.trySetProperty(name, value)
+    }
+
+    override fun getProperties(): MutableMap<String, *> {
+        onAccess(MemberKind.PROPERTY, null)
+        return delegate.properties
+    }
+
+    override fun getMissingProperty(name: String?): MissingPropertyException {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.getMissingProperty(name)
+    }
+
+    override fun setMissingProperty(name: String?): MissingPropertyException {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.setMissingProperty(name)
+    }
+
+    override fun methodMissingException(name: String?, vararg params: Any?): MissingMethodException {
+        onAccess(MemberKind.METHOD, name)
+        return delegate.methodMissingException(name, *params)
+    }
+
+    override fun getProperty(name: String?): Any {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.getProperty(name)
+    }
+
+    override fun setProperty(name: String?, value: Any?) {
+        onAccess(MemberKind.PROPERTY, name)
+        return delegate.setProperty(name, value)
+    }
+
+    override fun invokeMethod(name: String?, vararg arguments: Any?): Any {
+        onAccess(MemberKind.METHOD, name)
+        return delegate.invokeMethod(name, *arguments)
+    }
+
+    private
+    enum class MemberKind {
+        PROPERTY, METHOD
+    }
+
+    private
+    fun onAccess(memberKind: MemberKind, memberName: String?) {
+        val trace = run {
+            val location = userCodeContext.location(null)
+            when (memberKind) {
+                MemberKind.PROPERTY -> {
+                    if (memberName != null)
+                        PropertyTrace.Property(PropertyKind.PropertyUsage, memberName, PropertyTrace.Project(referrerProject.path, location))
+                    else location
+                }
+
+                // method lookup is more clear from the stack trace, so keep the minimal trace pointing to the location:
+                MemberKind.METHOD -> location
+            }
+        }
+
+        val message = StructuredMessage.build {
+            text("Project ")
+            reference(referrerProject.identityPath.toString())
+            text(" cannot dynamically lookup a ")
```

Comment:
"look up"

https://english.stackexchange.com/a/555267

Reply:
Fixed, thank you!